### PR TITLE
Matcher support for any objects inheriting `Comparable<T>`

### PR DESCRIPTION
**Pull request by hhandoko**
**NOTE:** A reimplementation for `release/2.0` branch from #122.

Added support for matchers comparing two objects inheriting Comparable<T>.

Test conditions reads closer to natural language, e.g. `value1 shouldBe gt(value2)` as opposed to `(value1 > value2) shouldBe true`.

### CODE REVIEW

File: src/main/kotlin/io/kotlintest/matchers/ComparableMatchers.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package io.kotlintest.matchers
+
+@Deprecated("Use `value shouldBe lt(x)` or `value should beLessThan(x)`")
```
Comment (martin-g) [inline src/main/kotlin/io/kotlintest/matchers/ComparableMatchers.kt]: What is the idea behind introducing a new class which methods are born deprecated ?

Comment (hhandoko) [inline src/main/kotlin/io/kotlintest/matchers/ComparableMatchers.kt]: Primarily keeping to conventions (as like `Int` or `Long` matchers), as they have the same surface behaviours. Also with the added benefit of helping users coming from other libraries and/or `v1`.

**Comment by sksamuel** [comment]
:+1:

**Comment by hhandoko** [comment]
Thanks!

**Comment by helmbold** [comment]
Do we really want to introduce *new* code annotated with `@Deprecated`?

**Comment by hhandoko** [comment]
@helmbold I'm not fussed if it was removed. I just thought it would provide some context to new KotlinTest users since the surface behaviours are the same with existing matchers.

**Comment by helmbold** [comment]
What do you think @sksamuel : should we relase a new version with deprecated methods? If I would be a solo developer I would remove them ...

**Comment by sksamuel** [comment]
Probably take them out.

**Comment by hhandoko** [comment]
OK, do you guys want me to remove it? If so, from which branch?

**Comment by sksamuel** [comment]
Release 2.

**Comment by hhandoko** [comment]
@sksamuel I just had a look at `release/2.0` branch and the `@Deprecated` methods for `Comparable<T>` matchers are not there. Did you already remove it?

**Comment by helmbold** [comment]
The deprecated methods were removed in commit [41854941f095e95fcc7f2bb92dc665260fe8d4a8](https://github.com/kotlintest/kotlintest/commit/41854941f095e95fcc7f2bb92dc665260fe8d4a8) by Aleix Morgadas.

**Comment by hhandoko** [comment]
OK, cool 👌

**Pull request merged**